theano/tensor/signal/downsample.py

```diff
@@ -343,17 +358,24 @@ def make_node(self, x, maxout, gz):
     def perform(self, node, inp, out):
         x, maxout, gz = inp
         gx_stg, = out
-        gx = numpy.zeros_like(x)
-
+        
         #number of pooling output rows
         pr = maxout.shape[-2]
         #number of pooling output cols
         pc = maxout.shape[-1]
         ds0, ds1 = self.ds
         st0, st1 = self.st
-        img_rows = x.shape[-2]
-        img_cols = x.shape[-1]
-
+        img_rows = x.shape[-2] + 2 * self.padding[0]
+        img_cols = x.shape[-1] + 2 * self.padding[1]
+        pad_h = self.padding[0]
+        pad_w = self.padding[1]
+        
+        
+        # pad the image
+        fill = x.min()-1
+        y = numpy.zeros((x.shape[0], x.shape[1], img_rows, img_cols)) + fill
```

Comment:
You have to give `y` the same `dtype` as `x`, otherwise the comparison later will not work.